File: CrudLayered_VB - MySql/Presentation/Forms/FormCliente.vb
Imports Domain

Public Class FormCliente
    Dim Cliente As New ClienteModel()

    Private Sub FormCliente_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        ListarEmpleados()
        Limpiar()
    End Sub

    Private Sub btnSave_Click(sender As Object, e As EventArgs) Handles btnSave.Click
        Cliente.Nombre = txtNombre.Text
        Cliente.Apellido = txtApellido.Text
        Cliente.Rfc = txtRfc.Text
        Cliente.Domicilio = txtDomicilio.Text


        Dim valid = New DataValidation(Cliente).validate()
        If valid = True Then
            Dim result = Cliente.saveChanges()
            MessageBox.Show(result)
            ListarEmpleados()
            Limpiar()
        End If
    End Sub

    Private Sub ListarEmpleados()
        dataGridView1.DataSource = Cliente.GetClientes()
    End Sub

    Private Sub Limpiar()
        txtNombre.Clear()
        txtApellido.Clear()
        txtRfc.Clear()
        txtDomicilio.Clear()
        panel1.Enabled = False
    End Sub

    Private Sub btnNew_Click(sender As Object, e As EventArgs) Handles btnNew.Click
        panel1.Enabled = True
    End Sub

    Private Sub btnEdit_Click(sender As Object, e As EventArgs) Handles btnEdit.Click
        If dataGridView1.SelectedRows.Count > 0 Then
            panel1.Enabled = True
            Cliente.ClienteId = dataGridView1.CurrentRow.Cells(0).Value
            Cliente.State = EntityState.Modified

            txtNombre.Text = dataGridView1.CurrentRow.Cells(1).Value
            txtApellido.Text = dataGridView1.CurrentRow.Cells(2).Value
            txtRfc.Text = dataGridView1.CurrentRow.Cells(3).Value
            txtDomicilio.Text = dataGridView1.CurrentRow.Cells(4).Value
        Else
            MessageBox.Show("Seleccione una fila")
        End If
    End Sub
End Class

File: Proyecto - Mysql/Presentacion/Forms/FormCliente.cs
using Domain.Models;
using Domain.ValueObjects;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Presentacion.Forms
{
    public partial class FormCliente : Form
    {
        private ClienteModel cliente = new ClienteModel();

        public FormCliente()
        {
            InitializeComponent();
            panel1.Enabled = true;
        }

        private void FormCliente_Load(object sender, EventArgs e)
        {
            ListarClientes();
        }

        private void ListarClientes()
        {
            dataGridView1.DataSource = cliente.GetAll();
        }

        private void btnSave_Click(object sender, EventArgs e)
        {
            cliente.Nombre = txtNombre.Text;
            cliente.Apellido = txtApellido.Text;
            cliente.Rfc = txtRfc.Text;
            cliente.Domicilio = txtDomicilio.Text;

            bool valid = new Helps.DataValidation(cliente).Validate();

            if (valid == true)
            {
                string result = cliente.saveChanges();
                MessageBox.Show(result);
                ListarClientes();
                Restart();
            }
        }


        private void Restart()
        {
            panel1.Enabled = false;
            txtNombre.Clear();
            txtApellido.Clear();
            txtDomicilio.Clear();
            txtRfc.Clear();
        }

        private void btnEdit_Click(object sender, EventArgs e)
        {
            if (dataGridView1.SelectedRows.Count > 0)
            {
                cliente.State = EntityState.Modified;
                cliente.ClienteId = Convert.ToInt32(dataGridView1.CurrentRow.Cells[0].Value);

                panel1.Enabled = true;
                txtNombre.Text = dataGridView1.CurrentRow.Cells[1].Value.ToString();
                txtApellido.Text = dataGridView1.CurrentRow.Cells[2].Value.ToString();
                txtRfc.Text = dataGridView1.CurrentRow.Cells[3].Value.ToString();
                txtDomicilio.Text = dataGridView1.CurrentRow.Cells[4].Value.ToString();
            }
            else
            {
                MessageBox.Show("seleccione una fila antes de modificar");
            }
        }

        private void btnNew_Click(object sender, EventArgs e)
        {
            panel1.Enabled = true;
            cliente.State = EntityState.Added;
        }
    }
}
